server/datastore/mysql/software_test.go

```diff
@@ -2929,3 +2931,437 @@ func testVerifySoftwareChecksum(t *testing.T, ds *Datastore) {
 		require.Equal(t, software[i], got)
 	}
 }
+
+func testListHostSoftware(t *testing.T, ds *Datastore) {
```

Comment:
That's a long test, but setting up the data is a big part so I opted to keep it all in one test. It should be readable from top to bottom though.